src/ucm/cuda/cudamem.c

```diff
@@ -402,17 +278,23 @@ static ucs_status_t ucm_cudamem_install(int events)
         goto out_unlock;
     }
 
-    for (patch = patches; patch->symbol != NULL; ++patch) {
-        status = ucm_reloc_modify(patch);
-        if (status != UCS_OK) {
-            ucm_warn("failed to install relocation table entry for '%s'", patch->symbol);
-            goto out_unlock;
+    status = ucm_cuda_install_hooks(ucm_cuda_driver_funcs, &used_reloc);
+    if (status != UCS_OK) {
+        ucm_warn("failed to install cuda memory hooks on driver API");
+    } else if (!used_reloc) {
+        ucm_debug("installed cuda memory bistro hooks on driver API");
+        ucm_cudamem_installed = 1;
+    } else if (status == UCS_OK) {
+        status = ucm_cuda_install_hooks(ucm_cuda_runtime_funcs, &used_reloc);
+        if (status == UCS_OK) {
+            ucm_debug("installed cuda memory %s hooks on runtime API",
+                      used_reloc ? "reloc" : "bistro");
```

Comment:
seems "ucm_cuda_install_hooks" can do bistro for some funcs if it is possible and reloc for others. is this debug msg accurate?